Assets/Scripts/Actor Components/Buff.cs

```diff
@@ -44,14 +49,36 @@ public void RemoveBuff()
             if (buffTimeout != null)
             {
                 StopCoroutine(buffTimeout);
+                StopCoroutine(blinkingTime);
                 buffTimeout = null;
+                blinkingTime = null;
             }
         }
     }
 
+    private void StartBlinking()
+    {
+        photonView.RPC("RPC_StartBlinking", RpcTarget.All);
+    }
+
     private IEnumerator ExpireBuff() 
     {
-        yield return new WaitForSeconds(buffDuration);
+        yield return new WaitForSeconds(Math.Max(0, buffDuration - blinkingDuration));
+        StartBlinking();
+    }
+
+    private IEnumerator ExpireBlinking()
+    {
+        int numOfBlinks = (int) Math.Floor(Math.Min(blinkingDuration, buffDuration) / blinkingPeriod);
```

Comment:
Good job on handling the case where the buff duration is too short/blinking duration is too long.